examples/api-samples/src/browser-only/filesystem/example-filesystem-initialization.ts

```diff
@@ -19,15 +19,15 @@ import { inject, injectable, interfaces } from '@theia/core/shared/inversify';
 import { EncodingService } from '@theia/core/lib/common/encoding-service';
 import { BrowserFSInitialization, DefaultBrowserFSInitialization } from '@theia/filesystem/lib/browser-only/browserfs-filesystem-initialization';
 import { BrowserFSFileSystemProvider } from '@theia/filesystem/lib/browser-only/browserfs-filesystem-provider';
-import type { FSModule } from 'browserfs/dist/node/core/FS';
+import { fs } from '@zenfs/core';
 
 @injectable()
 export class ExampleBrowserFSInitialization extends DefaultBrowserFSInitialization {
 
     @inject(EncodingService)
     protected encodingService: EncodingService;
 
-    override async initializeFS(fs: FSModule, provider: BrowserFSFileSystemProvider): Promise<void> {
+    override async initializeFS(provider: BrowserFSFileSystemProvider): Promise<void> {
```

Comment:
I did not look into zenfs. Is this the expected usage? Seems weird to get rid of the filesystem as a parameter as it seems to lose flexibility.

Reply:
@sdirix,

> Is this the expected usage? 

Yes, using the `fs` import/export is correct.

>  Seems weird to get rid of the filesystem as a parameter as it seems to lose flexibility.

This has simplified the internal API a lot since ZenFS doesn't need to worry about handling instances of the emulation layer. ZenFS' [`configure`](https://zenfs.dev/core/functions/configure.html) allows a lot of flexibility. 

Additionally, if you'd like to continue using dependency injection, you can still do so. For example:

```ts
import type { fs } from '@zenfs/core';

export type FSModule = typeof fs;

export class ExampleFSInitialization {
	// ...
	initializeFS(fs: FSModule) // ...
	// ...
}
```

However, I would not recommend this since it adds needed complexity. It may be easier to allow the user to pass in the configuration object instead.